vault_service/views/user.py

```diff
@@ -160,9 +160,21 @@ def store_data():
                 return '{}', 200 # or return 404?
             return q.user_data or '{}', 200
     elif request.method == 'POST':
-        d = json.dumps(payload)
+        # check if user record exists first
+        try:
+            with current_app.session_scope() as session:
+                q = session.query(User).filter_by(id=user_id).first()
+                data = json.loads(q.user_data)
+            for key in payload.keys():
+                # don't duplicate keys
+                data[key] = payload[key]
+            d = json.dumps(data)
+        except:
+            d = json.dumps(payload)
+
         if len(d) > MAX_ALLOWED_JSON_SIZE:
```

Comment:
Something to argue against and for: Should we have a list of valid keys that the user is allowed to store and then limit the max length of the values? This would be less flexible that what we have, but provides more control of what we store in the DB.

Reply:
I'm reluctant to limit the list of valid keys here, but I can implement a limit on the max length of the values.